GnuCash Help->GnuCash Manual and doc/lang->lang/doc directory structure

gjanssens opened the pull request:
* All build artefacts that were named _gnucash-help{.something}_ have been renamed to _gnucash-manual{.something}_
* Target names have been shortened. Wherever the target name contained _gnucash-help_ this has been replaced with _manual_. Similarly _gnucash-guide_ has been replaced with _guide_
* Source directories have been reshuffled. New structure is to have language directories at the top level. Each language directory can have one or more document directories (`guide|manual`).

This PR should be merged in sync with a related PR in the gnucash repository, the gnucash-on-macos repository and the gnucash-htdocs repository. The gnucash-on-windows and gnucash-on-flatpak repositories are not directly affected.

gjanssens committed:
Help Manual - rename all output formats from gnucash-help to gnucash-manual

gjanssens committed:
Remove redundant 'gnucash-' from target names

fellen reviewed (commented):
We will also need to review [Documentation Improvement](https://wiki.gnucash.org/wiki/Documentation_Improvement) etc.

gjanssens committed:
Remove 'gnucash-' prefix from examples in README

gjanssens commented:
> We will also need to review [Documentation Improvement](https://wiki.gnucash.org/wiki/Documentation_Improvement) etc.

Yeah, but perhaps that's best done when we start the beta releases and at the same time discourage any further work on maint. Otherwise we have to keep two sets of documentation improvement guidelines.

gjanssens commented on README:
```diff
@@ -153,8 +153,8 @@ depending on the output format you require.
   make check # for toplevel check
-  make de-gnucash-help-check # for a cmake check for one specific document
-                             # in this example - the German help guide
+  make de-gnucash-manual-check # for a cmake check for one specific document
```
Good catch.

gjanssens commented on README:
```diff
@@ -141,7 +141,7 @@ languages at once by omitting the language specifier in the target.
 For example:
 
-  make gnucash-help-html
+  make gnucash-manual-html
```
Fixed as well.

fellen commented:
> > We will also need to review [Documentation Improvement](https://wiki.gnucash.org/wiki/Documentation_Improvement) etc.
> 
> Yeah, but perhaps that's best done when we start the beta releases and at the same time discourage any further work on maint. Otherwise we have to keep two sets of documentation improvement guidelines.

You could already add it:
```
;From 4.900: <para 1>
: <para 2> …
;Up to 4.x, x<900: (Indent the old stuff by prepend paragraphs by a colon)

```
Then it does not get forgotten and current contributors learn about the difference.

gjanssens committed:
Restructure source directories from doc/lang to lang/doc

This is a better match for a editor's workflow.

Worth noting: this commit also changes intermediary-level targets
to be language based (C-check, de-pdf,...) instead of
document based (guide-check, manual-html,...)
The other targets are unaffected (C-guide-check, check,...)
